cpp/src/cider/tests/nextgen/compiler/DecimalOpTest.cpp

```diff
@@ -0,0 +1,146 @@
+  auto eu = substrait2eu.createRelAlgExecutionUnit();
+
+  // Pipeline Building
+  auto pipeline = parsers::toOpPipeline(eu);
+  transformer::Transformer transformer;
+  auto translators = transformer.toTranslator(pipeline);
+
+  // Codegen
+  context::CodegenContext codegen_ctx;
+  auto module = cider::jitlib::LLVMJITModule("test", true);
+  cider::jitlib::JITFunctionPointer function =
+      cider::jitlib::JITFunctionBuilder()
+          .registerModule(module)
+          .setFuncName("query_func")
+          .addReturn(cider::jitlib::JITTypeTag::VOID)
+          .addParameter(cider::jitlib::JITTypeTag::POINTER,
+                        "context",
+                        cider::jitlib::JITTypeTag::INT8)
+          .addParameter(cider::jitlib::JITTypeTag::POINTER,
+                        "input",
+                        cider::jitlib::JITTypeTag::INT8)
+          .addProcedureBuilder(
+              [&codegen_ctx, &translators](cider::jitlib::JITFunctionPointer func) {
+                codegen_ctx.setJITFunction(func);
+                translators->consume(codegen_ctx);
+                func->createReturn();
+              })
+          .build();
+  module.finish();
+
+  auto query_func = function->getFunctionPointer<void, int8_t*, int8_t*>();
+  // Execution
+  auto runtime_ctx = codegen_ctx.generateRuntimeCTX(allocator);
+  auto input_builder = ArrowArrayBuilder();
+  auto&& [schema, array] =
+      input_builder.setRowNum(operand_a.size())
+          .template addColumn<__int128_t>("a", CREATE_SUBSTRAIT_TYPE(Decimal), operand_a)
+          .template addColumn<__int128_t>("b", CREATE_SUBSTRAIT_TYPE(Decimal), operand_b)
+          .build();
+
+  query_func((int8_t*)runtime_ctx.get(), (int8_t*)array);
+
+  auto output_batch_array = runtime_ctx->getOutputBatch()->getArray();
+  auto output_batch_schema = runtime_ctx->getOutputBatch()->getSchema();
+
+  NativeType* data_buffer = (NativeType*)output_batch_array->children[0]->buffers[1];
+  const char* format_str = output_batch_schema->children[0]->format;
+  std::string::size_type sz;
+  int precision = std::stoi(&format_str[2], &sz);
+  int scale = std::stoi(&format_str[2 + sz + 1], &sz);
+  for (size_t i = 0; i < expected_c.size(); ++i) {
+    EXPECT_EQ(expected_c[i],
+              CiderInt128Utils::Decimal128ToString(data_buffer[i], precision, scale));
+  }
+}
+
+TEST_F(DecimalCompilerTest, DecimalArithmeticTest) {
+  std::string create_ddl =
+      "CREATE TABLE test(a DECIMAL(13,1) NOT NULL, b DECIMAL(13,2) NOT NULL);";
+  // operand unscaled value
+  std::vector<__int128_t> a_uv = {1, 2, 3, 4, 5};
+  std::vector<__int128_t> b_uv = {10, 20, 30, 40, 50};
+  // expected real value
+  std::vector<std::string> add_rv = {"0.20", "0.40", "0.60", "0.80", "1.00"};
+  std::vector<std::string> sub_rv = {"0.00", "0.00", "0.00", "0.00", "0.00"};
+  std::vector<std::string> mul_rv = {"0.010", "0.040", "0.090", "0.160", "0.250"};
+  std::vector<std::string> div_rv = {
```

Comment:
why division result precision is 5? it's decided by substrait plan?

Reply:
yes,  Decimal Arithmetic result type is determined by plan Arithmetic rules.
https://github.com/substrait-io/substrait/blob/0608878b25e7f9b4b3ffe33662eea9ef0f016548/extensions/functions_arithmetic_decimal.yaml#L66